src/pybind/mgr/dashboard/frontend/src/app/shared/services/user-change-password.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable } from '@angular/core';
+import { I18n } from '@ngx-translate/i18n-polyfill';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserChangePasswordService {
+  requiredPasswordRulesMessage: string;
+  passwordStrengthLevel: string;
+  passwordStrengthDescription: string;
+
+  constructor(private i18n: I18n) {}
+  getPasswordRulesMessage() {
+    return this.i18n(
+      'Required  rules for password complexity:\n\
+        - cannot contain username\n\
+        - cannot contain any keyword used in Ceph\n\
+        - must  consist of characters from the following groups:\n\
+            * alphabetic a-z, A-Z\n\
+            * numbers 0-9\n\
+            * special chars: !"#$%& \'()*+,-./:;<=>?@[\\]^_`{{|}}~\n\
+            * any other characters (signs)'
+    );
+  }
+
+  checkPasswordComplexity(password) {
+    this.passwordStrengthLevel = 'passwordStrengthLevel0';
+    this.passwordStrengthDescription = '';
+    const credits = this.checkPasswordComplexityLetters(password);
+    if (credits) {
+      if (credits < 10) {
+        this.passwordStrengthLevel = 'passwordStrengthLevel0';
+        this.passwordStrengthDescription = this.i18n('Too weak');
+      } else {
+        if (credits < 15) {
+          this.passwordStrengthLevel = 'passwordStrengthLevel1';
+          this.passwordStrengthDescription = this.i18n('Weak');
+        } else {
+          if (credits < 20) {
+            this.passwordStrengthLevel = 'passwordStrengthLevel2';
+            this.passwordStrengthDescription = this.i18n('OK');
+          } else {
+            if (credits < 25) {
+              this.passwordStrengthLevel = 'passwordStrengthLevel3';
+              this.passwordStrengthDescription = this.i18n('Strong');
+            } else {
+              this.passwordStrengthLevel = 'passwordStrengthLevel4';
+              this.passwordStrengthDescription = this.i18n('Very strong');
+            }
+          }
+        }
+      }
+    }
+    return [this.passwordStrengthLevel, this.passwordStrengthDescription];
+  }
+
+  private checkPasswordComplexityLetters(password): number {
+    const digitsNumber = password.replace(/[^0-9]/g, '').length;
+    const smallLettersNumber = password.replace(/[^a-z]/g, '').length;
+    const bigLettersNumber = password.replace(/[^A-Z]/g, '').length;
+    const punctuationNumber = password.replace(/[^!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~]/g, '').length;
+    const othersCharactersNumber =
+      password.length - (digitsNumber + smallLettersNumber + bigLettersNumber + punctuationNumber);
```

Comment:
Other characters would be ` ` (space) for example.
But to be more expressive about whats wrong with the password something like the following could be helpful:
```ts
const validatePw = pw => {
  return [
    Boolean(pw.match(/[0-9]/)), // Numbers
    Boolean(pw.match(/[a-z]/)), // Small letters
    Boolean(pw.match(/[A-Z]/)), // Capital letters
    Boolean(pw.match(/[^a-zA-Z0-9]/)) // Other characters
  ];
};

```